src/EPR.SubsidiaryBulkUpload.Application.UnitTests/Services/NotificationServiceTests.cs

```diff
@@ -100,4 +101,148 @@ public async Task SetErrorStatus_ShouldSerializeErrorsAndSetStringInRedisAndLogI
+        var key = "emptyKey";
+
+        _redisDatabaseMock.Setup(db => db.StringGetAsync(It.IsAny<RedisKey>(), It.IsAny<CommandFlags>()))
+            .ReturnsAsync(RedisValue.Null);
+
+        // Act
+        var result = await _notificationService.GetNotificationErrorsAsync(key);
+
+        // Assert
+        Assert.IsNotNull(result);
+        Assert.AreEqual(result.Errors.Count, 0);
+    }
+
+    [TestMethod]
+    public async Task GetNotificationErrorsAsync_ValidKey_ReturnsErrorsResponse()
+    {
+        // Arrange
+        var key = "validKey";
+        var errors = new UploadFileErrorResponse
+        {
+            Errors = new List<UploadFileErrorModel>
+                {
+                    new UploadFileErrorModel { FileLineNumber = 1, FileContent = "Content1", Message = "Message1", IsError = true, ErrorNumber = 6 },
+                    new UploadFileErrorModel { FileLineNumber = 2, FileContent = "Content2", Message = "Message2", IsError = false, ErrorNumber = 9 }
+                }
+        };
+
+        var json = JsonSerializer.Serialize(errors);
+        _redisDatabaseMock.Setup(db => db.StringGetAsync(It.IsAny<RedisKey>(), It.IsAny<CommandFlags>()))
+            .ReturnsAsync((RedisValue)json);
+
+        // Act
+        var result = await _notificationService.GetNotificationErrorsAsync(key);
+
+        // Assert
+        Assert.IsNotNull(result);
+        Assert.AreEqual(2, result.Errors.Count);
+        Assert.AreEqual(1, result.Errors[0].FileLineNumber);
+        Assert.AreEqual("Content1", result.Errors[0].FileContent);
+        Assert.AreEqual("Message1", result.Errors[0].Message);
+        Assert.AreEqual(6, result.Errors[0].ErrorNumber);
+        Assert.IsTrue(result.Errors[0].IsError);
+        Assert.AreEqual(2, result.Errors[1].FileLineNumber);
+        Assert.AreEqual("Content2", result.Errors[1].FileContent);
+        Assert.AreEqual("Message2", result.Errors[1].Message);
+        Assert.IsFalse(result.Errors[1].IsError);
+        Assert.AreEqual(9, result.Errors[1].ErrorNumber);
+
+        _loggerMock.Verify(
+            x => x.Log(
+                It.Is<LogLevel>(l => l == LogLevel.Information),
+                It.IsAny<EventId>(),
+                It.Is<It.IsAnyType>((v, t) => v.ToString() == $"Redis errors response key: {key} errors: {json}"),
+                It.IsAny<Exception>(),
+                It.Is<Func<It.IsAnyType, Exception, string>>((v, t) => true)),
+            Times.Once);
+    }
+
+    [TestMethod]
+    public async Task ClearRedisKeyAsync_KeyExists_ShouldLogKeyDeleted()
+    {
+        // Arrange
+        var key = "testKey";
+        _redisDatabaseMock
+            .Setup(db => db.KeyDeleteAsync(key, It.IsAny<CommandFlags>()))
+            .ReturnsAsync(true);
+
+        // Act
+        await _notificationService.ClearRedisKeyAsync(key);
+
+        // Assert
+        _redisDatabaseMock.Verify(db => db.KeyDeleteAsync(key, It.IsAny<CommandFlags>()), Times.Once);
+        _loggerMock.Verify(
```

Comment:
This can be simplified to
`_loggerMock.VerifyLog(x => x.LogInformation("Redis key {Key} deleted successfully.", key), Times.Once);`

Reply:
changed